perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java

```diff
@@ -705,6 +705,54 @@ public UserExtSource call(ApiCaller ac, Deserializer parms) throws PerunExceptio
 		}
 	},
 
+	/*#
+	 * Return userExtSource for specific attribute definition (specified by id) and unique value.
+	 * If not found, throw and exception.
+	 *
+	 * It looks for exactly one value of the specific attribute type:
+	 * - Integer -> exactly match
+	 * - String -> exactly match
+	 * - Map -> exactly match of "key=value"
+	 * - ArrayList -> exactly match of one of the value
+	 *
+	 * @param attributeId attribute id used for founding attribute definition which has to exists, be unique and in userExtSource namespace
+	 * @param attributeValue value used for searching
+	 * @return userExtSource found by attribute id and it's unique value
+	 */
+	/*#
+	 * Return userExtSource for specific attribute definition (specified by name) and unique value.
+	 * If not found, throw and exception.
+	 *
+	 * It looks for exactly one value of the specific attribute type:
+	 * - Integer -> exactly match
+	 * - String -> exactly match
+	 * - Map -> exactly match of "key=value"
+	 * - ArrayList -> exactly match of one of the value
+	 *
+	 * @param attributeName attribute name used for founding attribute definition which has to exists, be unique and in userExtSource namespace
+	 * @param attributeValue value used for searching
+	 * @return userExtSource found by attribute name and it's unique value
+	 */
+	getUserExtSourceByUniqueAttributeValue {
+
+		@Override
+		public UserExtSource call(ApiCaller ac, Deserializer parms) throws PerunException {
+			if (parms.contains("attributeId")) {
+				return ac.getUsersManager().getUserExtSourceByUniqueAttributeValue(
+					ac.getSession(),
+					parms.readInt("attributeId"),
+					parms.readString("attributeValue"));
+			} else if (parms.contains("attributeName")) {
```

Comment:
For simplification, you can use only `if` here.
```suggestion
			} 
			if (parms.contains("attributeName")) {
```

Reply:
Done